crates/red_knot_python_semantic/src/semantic_index/definition.rs

```diff
@@ -749,19 +754,40 @@ impl MatchPatternDefinitionKind {
 }
 
 #[derive(Clone, Debug)]
-pub struct ComprehensionDefinitionKind {
+pub struct ComprehensionDefinitionKind<'db> {
+    target_kind: TargetKind<'db>,
     iterable: AstNodeRef<ast::Expr>,
-    target: AstNodeRef<ast::ExprName>,
+    target: AstNodeRef<ast::Expr>,
```

Comment:
Unrelated to this PR but I think the `target` here is either a `ExprName` or `ExprAttribute`, so it might be useful to have an enum similar to `TargetKind` which has two variants like `TargetExpr::Name(ast::ExprName)` and `TargetExpr::Attribute(ast::ExprAttribute)`. This should not be done in this PR as it requires updating other attribute unpacking as well.

Reply:
OK, I'll work on this in a separate PR.

We also need to support `ExprSubscript` as a target expression.

Reply:
Just as a note it's not a priority to make this refactor, just something nice to have and is not necessarily required.